includes/core/classes/class-setup.php

```diff
@@ -165,21 +166,35 @@ public function filter_plugin_action_links( array $actions ): array {
 	}
 
 	/**
-	 * Activate the GatherPress plugin.
+	 * Activates the GatherPress plugin.
 	 *
-	 * This method performs activation tasks for the GatherPress plugin, such as renaming blocks and tables,
-	 * creating custom tables, and setting a flag to flush rewrite rules if necessary.
+	 * This method handles the activation of the GatherPress plugin. If the plugin
+	 * is being activated network-wide in a multisite installation, it iterates
+	 * through each blog in the network and performs necessary setup actions
+	 * (creating tables). If not network-wide, it only performs the setup actions
+	 * for the current site.
 	 *
 	 * @since 1.0.0
 	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param bool $network_wide Whether the plugin is being activated network-wide.
 	 * @return void
 	 */
-	public function activate_gatherpress_plugin(): void {
-		$this->maybe_create_custom_table();
-		$this->add_online_event_term();
+	public function activate_gatherpress_plugin( bool $network_wide ): void {
+		global $wpdb;
 
-		if ( ! get_option( 'gatherpress_flush_rewrite_rules_flag' ) ) {
-			add_option( 'gatherpress_flush_rewrite_rules_flag', true );
+		if ( is_multisite() && $network_wide ) {
+			// Get all blogs in the network and activate plugin on each one.
+			$blog_ids = $wpdb->get_col( $wpdb->prepare( 'SELECT blog_id FROM %i', $wpdb->blogs ) ); // phpcs:ignore WordPress.DB.PreparedSQLPlaceholders.UnsupportedIdentifierPlaceholder, WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
```

Comment:
Wouldn't it be easier to just use https://developer.wordpress.org/reference/functions/get_sites/

```php
$args = array(
	'fields' => 'ids',
	'network_id' => get_current_site()->id,
);
$blog_ids = get_sites( $args );
```

Reply:
There is no error, just the wish for core functions.